### PR TITLE
Getting accelerometer data

### PR DESCRIPTION
**Pull request by vgousyleblanc**
Looking for the acceleration data in the rootfile and converting it in csv

### CODE REVIEW

**Review by thomaslindner — APPROVED** [review]
Looks fine.

File: Makefile [hunk 1]
```diff
@@ -11,11 +11,11 @@ BINDIR= $(shell pwd)/bin
 VPATH = $(SRCDIR)
 
 CFLAGS=-c -g -Wall `root-config --cflags` -I${INCDIR}
-LDFLAGS=`root-config --glibs` -lHistPainter -lMinuit -L${ROOTSYS}/lib
+LDFLAGS=`root-config --glibs` -lHistPainter
```
Comment (pdeperio) [inline Makefile:14]: Was this on purpose? no other executables need these libraries?

File: Makefile [hunk 2]
```diff
@@ -11,11 +11,11 @@ BINDIR= $(shell pwd)/bin
...
 
 TARGET1=field_to_csv.cpp
 TARGET2=ptf_analysis.cpp
-TARGET3=ptf_ttree_analysis.cpp
+TARGET3=temperature_reading.cpp
```
Comment (pdeperio) [inline Makefile:18]: what happened to `ptf_ttree_analysis`?

File: README.md [hunk 3]
```diff
@@ -1,99 +1,68 @@
...
-+-- field_to_csv.cpp
-+-- ptf.config.dat                      # A configuration file that sets analysis options
-+-- ptf_analysis.cpp
-+-- ptf_charge_analysis.cpp
-+-- ptf_field_analysis.cpp
-+-- ptf_qe_analysis.cpp
-+-- ptf_timing_analysis.cpp
-+-- ptf_ttree_analysis.cpp
-```
+This is a simple C++ library for reading the ROOT files produced from the PTF. It handles loading the files and provides a simple way to access the data.
```
Comment (pdeperio) [inline README.md:3]: Seems like a big change to the README. Did you mean to remove all the top-level descriptions?

File: include/FindCircle.hpp [hunk 4]
```diff
@@ -21,7 +21,7 @@ struct Circle_st {
 /// gradhist that is built in this function.  Set any gradient values
 /// that are below cut_below * maximum_gradient to zero, then fit to
 /// circle. Returns Circle_st representing the best fit circle.
-Circle_st find_circle_max_grad( const TH2D* in, TH2D*& grad, double cut_below = 0.8 );
+Circle_st find_circle_max_grad( const TH2D* in, TH2D*& grad, double cut_below = 0.5 );
```
Comment (pdeperio) [inline include/FindCircle.hpp:24]: Non-accelerometer related? Should include such changes in a different PR, or include in the title/description of this one.

File: include/ScanPoint.hpp [hunk 5]
```diff
@@ -11,23 +11,28 @@
 class ScanPoint {
 public:
   ScanPoint(); // default constructor
-  void set_xyz( double x, double y, double z ){ fX=x; fY=y; fZ=z; }
+  void set_xyz( double x, double y, double z,double time_1,double t_ext2 ){ fX=x; fY=y; fZ=z;fT_ext2=t_ext2;
```
Comment (pdeperio) [inline include/ScanPoint.hpp:14]: Non-accelerometer related? Should include such changes in a different PR, or include in the title/description of this one.

File: include/temperature_correction.hpp [hunk 6]
```diff
@@ -0,0 +1,126 @@
+#include "WaveformFitResult.hpp"
```
Comment (pdeperio) [inline include/temperature_correction.hpp:1]: Non-accelerometer related? Should include such changes in a different PR, or include in the title/description of this one.

File: macros/collect_qe_plots.C [hunk 7]
```diff
@@ -150,11 +149,11 @@ TStyle* SetPTFStyle(Int_t WhichStyle = 1, TString styleName = "PTF") {
   // - inbuilt color schemes -
   // ptfStyle->SetPalette(1); // use the rainbow color set
   //ptfStyle->SetPalette(kViridis); // use the viridis color set
-  ptfStyle->SetPalette(kSunset); // use the sunset color set
+  //ptfStyle->SetPalette(kSunset); // use the sunset color set
```
Comment (pdeperio) [inline macros/collect_qe_plots.C:152]: Just changes in plotting script? Should include such changes in a different PR, or include in the title/description of this one.

File: ptf_qe_analysis.cpp [hunk 8]
```diff
@@ -2,19 +2,134 @@
 #include "ScanPoint.hpp"
 #include "Utilities.hpp"
 #include "FindCircle.hpp"
+#include "TDirectory.h"
```
Comment (pdeperio) [inline ptf_qe_analysis.cpp:5]: What are all the changes here? Should include such changes in a different PR, or include in the title/description of this one.

**Review by pdeperio — CHANGES_REQUESTED** [review]
In general, there seem to be a lot more modified files than the title/description of this PR would suggest is needed. It looks like you may have included several other developments you were working on locally. Best practice would be to isolate only the specific files/modifications you need to implement the feature you describe, and make separate PRs for other features/developments. (This PR also seems to be just a draft that you need to indicate is ready.)

File: Makefile (hunk 1 above)
Comment (vgousyleblanc) [reply Makefile:14]: No not on purpose, it is only that locally if I wanted to run this, I had to take it off I think. But you do need this library for the ptf_qe_analysis

File: Makefile (hunk 2 above)
Comment (vgousyleblanc) [reply Makefile:18]: I simply, did not wanted to create another target, that is what I will do !

File: README.md (hunk 3 above)
Comment (vgousyleblanc) [reply README.md:3]: Yes, probably a mistake, has to get back to the right version

File: include/FindCircle.hpp (hunk 4 above)
Comment (vgousyleblanc) [reply include/FindCircle.hpp:24]: Probably just a mistake, I will make these corrections !

File: include/ScanPoint.hpp (hunk 5 above)
Comment (vgousyleblanc) [reply include/ScanPoint.hpp:14]: Definately accelerometer related, it is how we get the values of the accelerometer in the ptf_analysis file

File: include/temperature_correction.hpp (hunk 6 above)
Comment (vgousyleblanc) [reply include/temperature_correction.hpp:1]: No significant change, just a test I was doing most probably

File: macros/collect_qe_plots.C (hunk 7 above)
Comment (vgousyleblanc) [reply macros/collect_qe_plots.C:152]: Probably a mistake, doesn't change anything in the accelerometer, no related.

File: ptf_qe_analysis.cpp (hunk 8 above)
Comment (vgousyleblanc) [reply ptf_qe_analysis.cpp:5]: Probably a test, no related to anything, it is definately not a part of the accelerometer script.

File: Makefile (hunk 1 above)
Comment (pdeperio) [reply Makefile:14]: OK, be careful when committing code to master, it should not break the system on `midptf`. Revert if necessary and commit directly to this branch. Or if you need, use some flags that determine what system you're compiling on to automatically set the necessary flags.

File: include/temperature_correction.hpp (hunk 6 above)
Comment (pdeperio) [reply include/temperature_correction.hpp:1]: OK, please remove anything unnecessary to the specific feature you're implementing in this PR. In general, you want to minimize the amount of code in each PR to facilitate review and debugging.

**Comment by pdeperio** [comment]
Also, 25 files changed seems like a lot for the (simple?) functionality you're implementing in this PR. Make sure to remove any files with unnecessary changes (or make a new PR with only the necessary files, if easier).

**Comment by vgousyleblanc** [comment]
So, I have now reverse all the needed changes. The reason why this commits as such a large number of files changes is that both the accelerometer data and temperature data are included in the same. There is also, a lot of different files used to modify something as simple as adding a variable. Finally, some changes are made to update the branch to the current master branch.

**Review by pdeperio — APPROVED** [review]

**Comment by pdeperio** [comment]
Sorry, forgot about this. Just merged. Please test `master` again.

**Pull request merged**